[Monolog] Added documentation about ElasticsearchLogstashHandler (#12201)
symfony/symfony-docs: logging/handlers.rst

Handlers
========

``ElasticsearchLogstashHandler``
--------------------------------

.. versionadded:: 4.4

    The ``ElasticsearchLogstashHandler`` was introduced in Symfony 4.4.

This handler dials directly with the HTTP interface of Elasticsearch. This means
it will slow down your application if Elasticsearch takes times to answer. Even
if all HTTP calls are done asynchronously.

In a development environment, it's fine to keep the default configuration: for
each log, an HTTP request will be made to push the log to Elasticsearch.

In a production environment, it's highly recommended to wrap this handler in a
handler with buffering capabilities (like the FingersCrossedHandler, or

> BufferHandler) in order to call Elasticsearch only once with a bulk push. For

Reviewer: ```suggestion
``BufferHandler``) in order to call Elasticsearch only once with a bulk push. For
```

btw. is it Buffer or Buffere**D** handler?

(https://www.elastic.co/what-is/elk-stack) stack is recommended.

To use it, declare it as a service:

.. configuration-block::

    .. code-block:: yaml

        # config/services.yaml
        services:
            Symfony\Bridge\Monolog\Handler\ElasticsearchLogstashHandler: ~

    .. code-block:: xml

        <!-- config/services.xml -->
        <?xml version="1.0" encoding="UTF-8" ?>
        <container xmlns="http://symfony.com/schema/dic/services"
            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
            xmlns:monolog="http://symfony.com/schema/dic/monolog"
            xsi:schemaLocation="http://symfony.com/schema/dic/services
                https://symfony.com/schema/dic/services/services-1.0.xsd
                http://symfony.com/schema/dic/monolog
                https://symfony.com/schema/dic/monolog/monolog-1.0.xsd">

            <services>
                <service id="Symfony\Bridge\Monolog\Handler\ElasticsearchLogstashHandler" />
            </services>
        </container>

    .. code-block:: php

        // config/services.php
        use App\Logger\SessionRequestProcessor;
        use Monolog\Formatter\LineFormatter;

>         $container->register('Symfony\\Bridge\\Monolog\\Handler\\ElasticsearchLogstashHandler');

Author: Sure :+1:

                    id: Symfony\Bridge\Monolog\Handler\ElasticsearchLogstashHandler

    .. code-block:: xml

        <!-- config/packages/prod/monolog.xml -->
        <?xml version="1.0" encoding="UTF-8" ?>
        <container xmlns="http://symfony.com/schema/dic/services"
            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
            xmlns:monolog="http://symfony.com/schema/dic/monolog"
            xsi:schemaLocation="http://symfony.com/schema/dic/services
                https://symfony.com/schema/dic/services/services-1.0.xsd
                http://symfony.com/schema/dic/monolog
                https://symfony.com/schema/dic/monolog/monolog-1.0.xsd">

            <monolog:config>
                <monolog:handler
                    name="es"
                    type="service"
                    id="Symfony\Bridge\Monolog\Handler\ElasticsearchLogstashHandler"
                />
            </monolog:config>
        </container>

    .. code-block:: php

        // config/packages/prod/monolog.php
        $container->loadFromExtension('monolog', [
            'handlers' => [
                'es' => [
                    'type' => 'service',

>                     'id' => 'Symfony\\Bridge\\Monolog\\Handler\\ElasticsearchLogstashHandler',

Reviewer: same question as above with the use statement.

Author: :+1:

Handlers
========

ElasticsearchLogstashHandler
----------------------------

.. versionadded:: 4.4

    The ``ElasticsearchLogstashHandler`` was introduced in Symfony 4.4.

> This handler dials directly with the HTTP interface of Elasticsearch. This means

Reviewer 2: This phrase looks strange to me. Maybe `dials with` should be `deals with`?

Author: I don't know. I'm not fluent :) I meant PHP dialog with ES via the HTTP interface. (You can communicate with ES with the "java interface").

Anyway, I updated the sentence with your proposal

    .. code-block:: yaml

        # config/services.yaml
        services:
            Symfony\Bridge\Monolog\Handler\ElasticsearchLogstashHandler: ~

    .. code-block:: xml

        <!-- config/services.xml -->
        <?xml version="1.0" encoding="UTF-8" ?>
        <container xmlns="http://symfony.com/schema/dic/services"
            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
            xmlns:monolog="http://symfony.com/schema/dic/monolog"
            xsi:schemaLocation="http://symfony.com/schema/dic/services
                https://symfony.com/schema/dic/services/services-1.0.xsd
                http://symfony.com/schema/dic/monolog
                https://symfony.com/schema/dic/monolog/monolog-1.0.xsd">

            <services>
                <service id="Symfony\Bridge\Monolog\Handler\ElasticsearchLogstashHandler" />
            </services>
        </container>

    .. code-block:: php

        // config/services.php
        use App\Logger\SessionRequestProcessor;
        use Monolog\Formatter\LineFormatter;
        use Symfony\Bridge\Monolog\Handler\ElasticsearchLogstashHandler;

        $container->register(ElasticsearchLogstashHandler::class);

> then use this referenced service in your monolog configuration:

Reviewer 2: `then` -> `Then`

Handlers
========

ElasticsearchLogstashHandler
----------------------------

.. versionadded:: 4.4

    The ``ElasticsearchLogstashHandler`` was introduced in Symfony 4.4.

This handler dials directly with the HTTP interface of Elasticsearch. This means
it will slow down your application if Elasticsearch takes times to answer. Even
if all HTTP calls are done asynchronously.

In a development environment, it's fine to keep the default configuration: for
each log, an HTTP request will be made to push the log to Elasticsearch.

In a production environment, it's highly recommended to wrap this handler in a
handler with buffering capabilities (like the ``FingersCrossedHandler`` or
``BufferHandler``) in order to call Elasticsearch only once with a bulk push. For
even better performance and fault tolerance, a proper `ELK stack`_ is recommended.

> To use it, declare it as a service:

Reviewer: ```suggestion
You need to declare it as a service first:
```

> 

Reviewer: ```suggestion

```

.. configuration-block::

    .. code-block:: yaml

        # config/services.yaml
        services:
            Symfony\Bridge\Monolog\Handler\ElasticsearchLogstashHandler: ~

    .. code-block:: xml

        <!-- config/services.xml -->
        <?xml version="1.0" encoding="UTF-8" ?>
        <container xmlns="http://symfony.com/schema/dic/services"
            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
            xmlns:monolog="http://symfony.com/schema/dic/monolog"
            xsi:schemaLocation="http://symfony.com/schema/dic/services
                https://symfony.com/schema/dic/services/services-1.0.xsd
                http://symfony.com/schema/dic/monolog
                https://symfony.com/schema/dic/monolog/monolog-1.0.xsd">

            <services>
                <service id="Symfony\Bridge\Monolog\Handler\ElasticsearchLogstashHandler" />
            </services>
        </container>

    .. code-block:: php

        // config/services.php
        use App\Logger\SessionRequestProcessor;
        use Monolog\Formatter\LineFormatter;

>         use Symfony\\Bridge\\Monolog\\Handler\\ElasticsearchLogstashHandler;

Reviewer: ```suggestion
        use Symfony\Bridge\Monolog\Handler\ElasticsearchLogstashHandler;
```

        $container->register(ElasticsearchLogstashHandler::class);

> then reference it in monolog configuration:

Reviewer: ```suggestion
then use this referenced service in your monolog configuration:
```

> 

Reviewer: ```suggestion

```

.. configuration-block::

    .. code-block:: yaml

        # config/packages/prod/monolog.yaml
        monolog:
            handlers:
                es:
                    type: service
                    id: Symfony\Bridge\Monolog\Handler\ElasticsearchLogstashHandler

    .. code-block:: xml

        <!-- config/packages/prod/monolog.xml -->
        <?xml version="1.0" encoding="UTF-8" ?>
        <container xmlns="http://symfony.com/schema/dic/services"
            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
            xmlns:monolog="http://symfony.com/schema/dic/monolog"
            xsi:schemaLocation="http://symfony.com/schema/dic/services
                https://symfony.com/schema/dic/services/services-1.0.xsd
                http://symfony.com/schema/dic/monolog
                https://symfony.com/schema/dic/monolog/monolog-1.0.xsd">

            <monolog:config>
                <monolog:handler
                    name="es"
                    type="service"
                    id="Symfony\Bridge\Monolog\Handler\ElasticsearchLogstashHandler"
                />
            </monolog:config>
        </container>

    .. code-block:: php

>         use Symfony\\Bridge\\Monolog\\Handler\\ElasticsearchLogstashHandler;

Reviewer: ```suggestion

```

>         // config/packages/prod/monolog.php

Reviewer: ```suggestion
        // config/packages/prod/monolog.php
       use Symfony\Bridge\Monolog\Handler\ElasticsearchLogstashHandler;        
```
